configuration/decode_clusters.go

```diff
@@ -0,0 +1,153 @@
+package configuration
+
+import (
+	"reflect"
+	"regexp"
+	"strconv"
+
+	"github.com/fabric8-services/fabric8-cluster/cluster/repository"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+func decodeClusters(data interface{}) ([]repository.Cluster, error) {
```

Comment:
Can we add few tests for  `decodeClusters`?

Reply:
sure, let me add some tests here as well (although, we have some coverage once the config is loaded, but that won't hurt to test more cases here)

Reply:
addressed in 10666f8